bubblesub/ui/menu.py

```diff
@@ -41,13 +41,36 @@ def _window_from_menu(menu: QtWidgets.QMenu) -> QtWidgets.QWidget:
     return window
 
 
-def _on_menu_about_to_show(log_api: LogApi, menu: QtWidgets.QMenu) -> None:
+def _toggle_action(
+    main_window: QtWidgets.QMainWindow, action: QtWidgets.QAction
+) -> None:
+    for cmd in action.commands:
+        if "show-widget" in cmd.names:
+            widget_name = str(cmd.args.target)
+            widget = main_window.findChild(QtWidgets.QWidget, widget_name)
+
+            if cmd.args.all:
+                widget = widget.parentWidget()
+
+            print_widget_name = widget_name.replace("-", " ")
+
+            if widget.isVisible():
+                action.setText("{}{}".format("Hide ", print_widget_name))
+            else:
+                action.setText("{}{}".format("Show ", print_widget_name))
```

Comment:
That's a very low quality hack. Please leave the text as `toggle` and in the future we can discuss dynamic menu labels

Reply:
Hmm, I know, it's not a good code, but can we leave it as-is for now and iterate over that eventually? I mean, it works for now, it's only horrible

Reply:
This is an open source project, so we can afford not checking in bad solutions even if they do work

Reply:
Okay, let's iterate over it later. Let's open an issue to record this new feature though